src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -697,14 +691,15 @@ object OrderedRVD {
     consumerRegion: Region,
     producerRegion: Region,
     ctx: RVDContext,
-    typ: OrderedRVDType,
+    oldType: OrderedRVDType,
```

Comment:
Are there constraints on what oldType and newType can be in relation to each other?

Reply:
Yeah, I missed that. Taking two types that had to be mostly the same was a bad interface. I've refactored it a bit and added some `requires` clauses.